### DISCUSSION TITLE
Minio subfolder creation

### DISCUSSION THREAD

**Discussion by techpet** [nhost/hasura-storage#148] [Q&A]
Is there a way to upload a file to subfolder inside the root minio folder using the nhost sdk? The [upload](https://docs.nhost.io/reference/javascript/storage/upload) function does not seem to provide such a parameter. Am I missing something from the documentation here?

**Comment by dbarrosop** [comment]
The nhost sdk doesn't upload files to minio, it uploads file to hasura-storage (which happens to use minio in this case). What is it you are trying to do exactly?

**Comment by techpet** [comment]
Exactly @dbarrosop ,thanks for the clarification. In my architecture there is a Workspace entity and I want to have each Workspace having a different bucketId and a different subfolder with it's name (or it's id) inside the hasura-storage. Then, when uploading a new file related to that specific Workspace, I want to associate this with the specific bucketId and have it stored inside the subfolder corresponding to that Workspace.

 Is that possible?

**Reply by dbarrosop** [reply]
No, that's not possible. `hasura-storage` doesn't support uploading files to different folders. If you really need to do that you are going to need different instances of `hasura-storage`. In any case, you should treat all of that like a black box, users don't have access to the underlying storage, only to the metadata and the files themselves so how they are arranged internally in minio shouldn't matter much.
